### PR TITLE
LPD-40881 Enable virtual items with no shipping address to be processed via Paypal

### PR DESCRIPTION
**Pull request by ghost**
https://liferay.atlassian.net/browse/LPD-40881

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentIntegration.java
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentIntegration.java:1]: All good, but I see we are not managing right the region at line 731 (743 on the new file) where we send null if the region is null. That will not work as we allow for null region in our addresses. Only the country is mandatory.
Can you please fix it? Thanks

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentIntegration.java [hunk 1]
```diff
@@ -683,6 +687,14 @@ private PayPalHttpClient _getPayPalHttpClient(
 				payPalGroupServiceConfiguration.clientSecret()));
 	}
 
+	private String _getRegionCode(Region region) {
+		if (region == null) {
+			return null;
```
[the remark this answers is not in the document]
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentIntegration.java:692]: Hi @luca-pellizzon I ran tests with and without shipping address and had no problems

**Comment by luca-pellizzon** [comment]
ci:forward:force

**Pull request closed without merging**